eth/vm/computation.py

```diff
@@ -125,7 +106,8 @@ class BaseComputation(Configurable, ABC):
     def __init__(self,
                  state: BaseState,
                  message: Message,
-                 transaction_context: BaseTransactionContext) -> None:
+                 transaction_context: BaseTransactionContext,
+                 tracer: BaseTracer = None) -> None:
```

Comment:
I've been struggling with whether the `tracer` should be a required parameter and I'm leaning towards yes.  My thought is that I already had a situation where my tests were failing because I *thought* I was using a real tracer but a code path hadn't been updated properly so it was defaulting to the `NoopTracer`.  By making this a required argument we can't make that mistake.

Reply:
@pipermerriam, I've made it required. As per tests - do you think keeping expected JSON-trace and then simply comparing it with a result will be a good test for that?

p.s. py35-lint went crazy lol - it blames to one line which passes checks in master... 
It behaves really strange, e.g. we have a lot of formatted strings like f"...", which "...are only supported in Python 3.6 and greater", how does master passes py35-lint checks then?

Reply:
The `eth` module still has to support python 3.5 so we don't use f-strings there.

Reply:
@pipermerriam, so why py35-lint is launching on a trinity module then? It fails on:
```__version__: str``` at the top of __init__.py.
It somehow passes same on master branch...

Reply:
Ah! `py35-lint` should **not** be run on the trinity module.  That is an error/mistake.

Reply:
@pipermerriam. Weird, I don't see any diff for tox.ini in master\tracer branches, but somehow master passes all tests, and tracer fails one py35-lint test here.